### ISSUE TITLE
Add guidance on how to resolve fatal git error

### ISSUE THREAD

**Issue by Thomas-Hirsch** [moj-analytical-services/user-guidance#10]
Several people have encountered an issue when switching branches: `fatal: index file smaller than expected`. This occurs when the index file gets corrupted, and can be fixed with 
```
$ rm .git/index
$ git reset HEAD .
```

Would be useful to add this in an FAQ section of the git guidance.

**Comment by ben-lavelle** [comment]
I think an improvement due to @r-mccormack was using an easy git add in the middle to make sure the index is rebuilt in between. Also worth having guidance on tidying up the staging area when it gets confused (select all and Revert--will write something longer if you want). So we have e.g.

```bash
$ rm .git/index
$ git add .
$ git reset HEAD
```

**Comment by r-mccormack** [comment]
Hello.

The `git add .` helps, I don't know why, but I theorise that in some cases it is keeping some sort of cached version of the old dodgy index file which it only gets rid off when you actually overwrite the file with a new one. 

That said `add . `is not a superb way of doing this, since it adds everything. Firstly, that means it takes ages, in some cases and secondly if for some reason they forgot to `git reset HEAD` afterwards there is the possibility they would commit something they didn't intend to, such as data.

We do it this way because that is guaranteed to work for everyone, but if anyone has a better suggestion of how to generate a clean index file before resetting that might be better.

**Comment by ben-lavelle** [comment]
Yeah I think the conversation we had, Richard, was that ideally you'd `add` just a specific (small) file that is always present and is uncontroversial etc. but for universalisability and copy-pasteability we couldn't think of a way to do that (another option we though of was adding a fourth line to create such a dummy file, but that seems a bit messy)

**Comment by r-mccormack** [comment]
Yes, the advantage of `git add .` is that it is clean, elegant and will work for everyone.

Personally I think the risks are low, but I felt I should make them known so that an informed decision can be made on whether to include this or not (and who knows, someone else might have a nicer solution!).  :)

**Comment by neillturner** [comment]
fixed by Various documentation fixes for outstanding Issues #193
